CourseGrab/AppDelegate.swift

```diff
@@ -95,12 +95,17 @@ extension AppDelegate: UNUserNotificationCenterDelegate {
       didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
         let tokenParts = deviceToken.map { data in String(format: "%02.2hhx", data) }
         let token = tokenParts.joined()
-        NetworkManager.shared.sendDeviceToken(deviceToken: token).observe { result in
-            switch result {
-            case .value:
-                break
-            case .error(let error):
-                print(error)
+        NetworkManager.shared.sendDeviceToken(deviceToken: token).chained { deviceTokenResponse -> Future<SuccessResponse> in
+            return NetworkManager.shared.enableNotifications(enabled: deviceTokenResponse.success)
+            }.observe { result in
+            DispatchQueue.main.async {
+                switch result {
+                case .value(let response):
+                    UserDefaults.standard.areNotificationsEnabled = response.success
+                    AppDevAnalytics.shared.logFirebase(MobileAlertPressPayload())
+                case .error(let error):
+                    print(error)
+                }
```

Comment:
I'm pretty sure we don't want this here. Registering for notifications is very different from enabling notifications. Registering tells iOS that we may or may not be receiving APNs, and it gives us a device token just in case. Enabling notification is how we notify the server of the user's preferences.